packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1361,6 +1533,17 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     if (isObject) {
       switch (newChild.$$typeof) {
         case REACT_ELEMENT_TYPE:
+          // This function recurses only on a top-level fragment,
+          // so that it is treated as a set of children
+          // Otherwise, we follow the normal flow.
+          if (newChild.type === REACT_FRAGMENT_TYPE && newChild.key === null) {
+            return reconcileChildFibersForTopLevelFragment(
```

Comment:
I actually think you don't even need to fork anything. You just need to call `updateSlot` here instead.